src/execution/sql/storage_interface.cpp

```diff
@@ -82,4 +84,13 @@ void StorageInterface::IndexDelete(storage::TupleSlot table_tuple_slot) {
   curr_index_->Delete(exec_ctx_->GetTxn(), *index_pr_, table_tuple_slot);
 }
 
+bool StorageInterface::IndexCreateInsert(catalog::index_oid_t index_oid, storage::TupleSlot table_tuple_slot) {
+  terrier::storage::index::IndexBuilder index_builder;
+  index_builder.SetSqlTableAndTransactionContext(exec_ctx_->GetTxn(), table_,
+                                                 exec_ctx_->GetAccessor()->GetIndexSchema(index_oid));
+  auto result = index_builder.Insert(exec_ctx_->GetAccessor()->GetIndex(index_oid), table_tuple_slot);
+  exec_ctx_->GetAccessor()->SetIndexLive(index_oid);
```

Comment:
This looks like you're setting the index live for every insert. Also see previous comments about index liveness.